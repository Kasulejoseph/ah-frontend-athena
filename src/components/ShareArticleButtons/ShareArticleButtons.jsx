import React from "react";
import {
  FacebookShareButton,
  TwitterShareButton,
  EmailShareButton,
  FacebookIcon,
  TwitterIcon,
  EmailIcon
} from "react-share";

export const ShareArticleButtons = () => (
  <div>
    <FacebookShareButton
      className="share-button"
      url={window.location.href}
      style={{ marginBottom: "8px" }}
    >
      <FacebookIcon size={32} round />
    </FacebookShareButton>
    <TwitterShareButton
      className="share-button"
      url={window.location.href}
      style={{ marginBottom: "8px" }}
    >
      <TwitterIcon size={32} round />
    </TwitterShareButton>
    <EmailShareButton
      className="share-button"
      url={window.location.href}
      style={{ marginBottom: "8px" }}
    >
      <EmailIcon size={32} round />
    </EmailShareButton>
  </div>
);

export default ShareArticleButtons;
