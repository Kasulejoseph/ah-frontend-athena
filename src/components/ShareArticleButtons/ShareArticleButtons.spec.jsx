import { shallow } from "enzyme";
import React from "react";
import { ShareArticleButtons } from "./ShareArticleButtons";

describe("Share article buttons tests", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<ShareArticleButtons />);
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should render 3 share buttons", () => {
    expect(wrapper.find(".share-button").length).toBe(3);
  });
});
