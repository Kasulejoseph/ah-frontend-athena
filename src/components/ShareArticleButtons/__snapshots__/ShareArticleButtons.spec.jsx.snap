// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Share article buttons tests should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShareArticleButtons />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <CreatedButton
          className="share-button"
          style={
            Object {
              "marginBottom": "8px",
            }
          }
          url="http://localhost/"
          windowHeight={400}
          windowWidth={550}
        >
          <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />
        </CreatedButton>,
        <CreatedButton
          className="share-button"
          style={
            Object {
              "marginBottom": "8px",
            }
          }
          url="http://localhost/"
          windowHeight={400}
          windowWidth={550}
        >
          <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />
        </CreatedButton>,
        <CreatedButton
          className="share-button"
          onClick={[Function]}
          openWindow={false}
          style={
            Object {
              "marginBottom": "8px",
            }
          }
          url="http://localhost/"
        >
          <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />
        </CreatedButton>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />,
          "className": "share-button",
          "style": Object {
            "marginBottom": "8px",
          },
          "url": "http://localhost/",
          "windowHeight": 400,
          "windowWidth": 550,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "iconBgStyle": Object {},
            "logoFillColor": "white",
            "round": true,
            "size": 32,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />,
          "className": "share-button",
          "style": Object {
            "marginBottom": "8px",
          },
          "url": "http://localhost/",
          "windowHeight": 400,
          "windowWidth": 550,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "iconBgStyle": Object {},
            "logoFillColor": "white",
            "round": true,
            "size": 32,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Icon
            className=""
            iconBgStyle={Object {}}
            logoFillColor="white"
            round={true}
            size={32}
          />,
          "className": "share-button",
          "onClick": [Function],
          "openWindow": false,
          "style": Object {
            "marginBottom": "8px",
          },
          "url": "http://localhost/",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "",
            "iconBgStyle": Object {},
            "logoFillColor": "white",
            "round": true,
            "size": 32,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <CreatedButton
            className="share-button"
            style={
              Object {
                "marginBottom": "8px",
              }
            }
            url="http://localhost/"
            windowHeight={400}
            windowWidth={550}
          >
            <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />
          </CreatedButton>,
          <CreatedButton
            className="share-button"
            style={
              Object {
                "marginBottom": "8px",
              }
            }
            url="http://localhost/"
            windowHeight={400}
            windowWidth={550}
          >
            <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />
          </CreatedButton>,
          <CreatedButton
            className="share-button"
            onClick={[Function]}
            openWindow={false}
            style={
              Object {
                "marginBottom": "8px",
              }
            }
            url="http://localhost/"
          >
            <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />
          </CreatedButton>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />,
            "className": "share-button",
            "style": Object {
              "marginBottom": "8px",
            },
            "url": "http://localhost/",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />,
            "className": "share-button",
            "style": Object {
              "marginBottom": "8px",
            },
            "url": "http://localhost/",
            "windowHeight": 400,
            "windowWidth": 550,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Icon
              className=""
              iconBgStyle={Object {}}
              logoFillColor="white"
              round={true}
              size={32}
            />,
            "className": "share-button",
            "onClick": [Function],
            "openWindow": false,
            "style": Object {
              "marginBottom": "8px",
            },
            "url": "http://localhost/",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "",
              "iconBgStyle": Object {},
              "logoFillColor": "white",
              "round": true,
              "size": 32,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
